src/foam/u2/wizard/util/CircleIndicator.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.wizard.util',
+  name: 'CircleIndicator',
+  extends: 'foam.u2.Element',
+  documentation: `
+    Displays a coloured circle containing a number or icon.
+  `,
+
+  requires: [
+    'foam.core.ExpressionSlot'
+  ],
+
+  css: `
+    ^ {
+      position: relative;
+      border-radius: 50%;
+      text-align: center;
+      display: inline-block;
+    }
+    ^ > img {
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      pointer-events: none;
+      z-index: -1;
+    }
+  `,
+
+  properties: [
+    // Configuration
+    {
+      name: 'label',
+      class: 'String'
+    },
+    {
+      name: 'borderColor',
+      class: 'String'
+    },
+    {
+      name: 'borderColorHover',
+      class: 'String'
+    },
+    {
+      name: 'textColor',
+      class: 'String',
+      expression: function (stateBorderColor_) { return stateBorderColor_; }
+    },
+    {
+      name: 'backgroundColor',
+      class: 'String'
+    },
+    {
+      name: 'borderThickness',
+      class: 'Int'
+    },
+    {
+      name: 'icon',
+      class: 'Image'
+    },
+    {
+      name: 'size',
+      class: 'Int',
+      value: 30
+    },
+
+    // State
+    {
+      name: 'hasMouseOver',
+      class: 'Boolean',
+      value: false
+    },
+    {
+      name: 'stateBorderColor_',
+      expression: function ( borderColor, borderColorHover, hasMouseOver ) {
+        return hasMouseOver ? borderColorHover : borderColor;
+      }
+    },
+    {
+      name: 'clickable',
+      class: 'Boolean',
+      value: false
+    }
+  ],
+
+  methods: [
+    function initE() {
+      this
+        .addClass(this.myClass())
+        .style({
+          'background-color': this.backgroundColor,
+          'border-color': this.stateBorderColor_$,
+          'width': '' + this.size + 'px',
+          'height': '' + this.size + 'px',
+          'line-height': '' + this.size + 'px',
+          'font-size': this.size * 0.65,
+          'color': this.textColor$,
+          'border': '' + this.borderThickness + 'px solid',
+          'cursor': this.ExpressionSlot.create({
+            obj: this,
+            code: function (clickable) {
+              return clickable ? 'pointer' : 'default';
+            }
+          }),
+        })
+        .on('mouseover', () => {
```

Comment:
Would it be easier to just use the :hover CSS selector?

Reply:
Not sure if I can use a selector here since I want the colour properties attached to slots. For example you can use CircleIndicator as a button in which case you may want the `:hover` colour to be the same as the default if the button turns disabled.